### DISCUSSION TITLE
how to use foocus with gradio client api

### DISCUSSION THREAD

**Discussion by ishangrotra** [lllyasviel/Fooocus#2613] [Q&A] [answered]
hey so i have been trying to use foocus with a gradio client api but but i always seem to get the error 
'Could not find \'data\' key in response. Response received: {\'detail\': [{\'type\': \'model_attributes_type\', \'loc\': [\'body\'], \'msg\': \'Input should be a valid dictionary or object to extract fields from\', \'input\': \'{"data": [null], "fn_index": 0, "session_hash": "7642beb0-f999-42f7-83aa-03483b560420"}\', \'url\': \'[https://errors.pydantic.dev/2.6/v/model_attributes_type\](https://errors.pydantic.dev/2.6/v/model_attributes_type/)'}]}'
can any one help me and tell me what is going wrong 
code used 
from gradio_client import Client

client = Client("link of the hosted gradio app")
result = client.predict(
				"ImagePrompt",	# str in 'Type' Radio component
				fn_index=2
)
print(result)

**Comment by t1hq** [comment, marked as answer]
so i'm not the only one ：）, I'm suffering this issue for a week, and then founded, I'm not python expert so I thought this might not be that hard.
the key api in focus 2.3.0 is `fn 40`, and `fn 41`:
- 40 sets all kinds of params
- 41 makes fooocus to draw

first call 40, then 41, done.
```python
client = Client("http://127.0.0.1:7865/")
result = client.predict(
	...
	fn_index=40
)
print(result)
result= client.predict(fn_index=41)
```
set params according to values in UI, not in sample.
a demo code is posted [here](https://tricks.one/post/get-fooocus-gradio-api-working-for-uov/), hopes chinese is not a problem.

**Reply by ishangrotra** [reply]
Hey thanks for replying on the gradio link ? for example https://4dc12345678.gradio.live/ also please correct me if i am wrong so i have to 1st set index to 40 then to 41 correct will it not work on index=2 ? since when i am trying to do index=41 i am getting a different error 
example code in documentation 
from gradio_client import Client

client = Client("https://4dc67e06d224931498.gradio.live/")
result = client.predict(
				True,	# bool in 'Generate Image Grid for Each Batch' Checkbox component
				"hello",	# str in 'parameter_11' Textbox component
				"",	# str in 'Negative Prompt' Textbox component
				["Fooocus V2"],	# List[str] in 'Selected Styles' Checkboxgroup component
				"Quality",	# str in 'Performance' Radio component
				"704×1408 <span style="color: grey;"> ∣ 1:2</span>",	# str in 'Aspect Ratios' Radio component
				1,	# int | float (numeric value between 1 and 32)
								in 'Image Number' Slider component
				"png",	# str in 'Output Format' Radio component
				"Howdy!",	# str in 'Seed' Textbox component
				True,	# bool in 'Read wildcards in order' Checkbox component
				0,	# int | float (numeric value between 0.0 and 30.0)
								in 'Image Sharpness' Slider component
				1,	# int | float (numeric value between 1.0 and 30.0)
								in 'Guidance Scale' Slider component
				"null",	# str (Option from: [])
								in 'Base Model (SDXL only)' Dropdown component
				"None",	# str (Option from: ['None'])
								in 'Refiner (SDXL or SD 1.5)' Dropdown component
				0.1,	# int | float (numeric value between 0.1 and 1.0)
								in 'Refiner Switch At' Slider component
				True,	# bool in 'Enable' Checkbox component
				"None",	# str (Option from: ['None'])
								in 'LoRA 1' Dropdown component
				-2,	# int | float (numeric value between -2 and 2)
								in 'Weight' Slider component
				True,	# bool in 'Enable' Checkbox component
				"None",	# str (Option from: ['None'])
								in 'LoRA 2' Dropdown component
				-2,	# int | float (numeric value between -2 and 2)
								in 'Weight' Slider component
				True,	# bool in 'Enable' Checkbox component
				"None",	# str (Option from: ['None'])
								in 'LoRA 3' Dropdown component
				-2,	# int | float (numeric value between -2 and 2)
								in 'Weight' Slider component
				True,	# bool in 'Enable' Checkbox component
				"None",	# str (Option from: ['None'])
								in 'LoRA 4' Dropdown component
				-2,	# int | float (numeric value between -2 and 2)
								in 'Weight' Slider component
				True,	# bool in 'Enable' Checkbox component
				"None",	# str (Option from: ['None'])
								in 'LoRA 5' Dropdown component
				-2,	# int | float (numeric value between -2 and 2)
								in 'Weight' Slider component
				True,	# bool in 'Input Image' Checkbox component
				"Howdy!",	# str in 'parameter_91' Textbox component
				"Disabled",	# str in 'Upscale or Variation:' Radio component
				"https://raw.githubusercontent.com/gradio-app/gradio/main/test/test_files/bus.png",	# str (filepath or URL to image)
								in 'Drag above image to here' Image component
				["Left"],	# List[str] in 'Outpaint Direction' Checkboxgroup component
				"https://raw.githubusercontent.com/gradio-app/gradio/main/test/test_files/bus.png",	# str (filepath or URL to image)
								in 'Drag inpaint or outpaint image to here' Image component
				"Howdy!",	# str in 'Inpaint Additional Prompt' Textbox component
				"https://raw.githubusercontent.com/gradio-app/gradio/main/test/test_files/bus.png",	# str (filepath or URL to image)
								in 'Mask Upload' Image component
				True,	# bool in 'Disable Preview' Checkbox component
				True,	# bool in 'Disable Intermediate Results' Checkbox component
				True,	# bool in 'Disable seed increment' Checkbox component
				0.1,	# int | float (numeric value between 0.1 and 3.0)
								in 'Positive ADM Guidance Scaler' Slider component
				0.1,	# int | float (numeric value between 0.1 and 3.0)
								in 'Negative ADM Guidance Scaler' Slider component
				0,	# int | float (numeric value between 0.0 and 1.0)
								in 'ADM Guidance End At Step' Slider component
				1,	# int | float (numeric value between 1.0 and 30.0)
								in 'CFG Mimicking from TSNR' Slider component
				"euler",	# str (Option from: ['euler', 'euler_ancestral', 'heun', 'heunpp2', 'dpm_2', 'dpm_2_ancestral', 'lms', 'dpm_fast', 'dpm_adaptive', 'dpmpp_2s_ancestral', 'dpmpp_sde', 'dpmpp_sde_gpu', 'dpmpp_2m', 'dpmpp_2m_sde', 'dpmpp_2m_sde_gpu', 'dpmpp_3m_sde', 'dpmpp_3m_sde_gpu', 'ddpm', 'lcm', 'ddim', 'uni_pc', 'uni_pc_bh2'])
								in 'Sampler' Dropdown component
				"normal",	# str (Option from: ['normal', 'karras', 'exponential', 'sgm_uniform', 'simple', 'ddim_uniform', 'lcm', 'turbo'])
								in 'Scheduler' Dropdown component
				-1,	# int | float (numeric value between -1 and 200)
								in 'Forced Overwrite of Sampling Step' Slider component
				-1,	# int | float (numeric value between -1 and 200)
								in 'Forced Overwrite of Refiner Switch Step' Slider component
				-1,	# int | float (numeric value between -1 and 2048)
								in 'Forced Overwrite of Generating Width' Slider component
				-1,	# int | float (numeric value between -1 and 2048)
								in 'Forced Overwrite of Generating Height' Slider component
				-1,	# int | float (numeric value between -1 and 1.0)
								in 'Forced Overwrite of Denoising Strength of "Vary"' Slider component
				-1,	# int | float (numeric value between -1 and 1.0)
								in 'Forced Overwrite of Denoising Strength of "Upscale"' Slider component
				True,	# bool in 'Mixing Image Prompt and Vary/Upscale' Checkbox component
				True,	# bool in 'Mixing Image Prompt and Inpaint' Checkbox component
				True,	# bool in 'Debug Preprocessors' Checkbox component
				True,	# bool in 'Skip Preprocessors' Checkbox component
				1,	# int | float (numeric value between 1 and 255)
								in 'Canny Low Threshold' Slider component
				1,	# int | float (numeric value between 1 and 255)
								in 'Canny High Threshold' Slider component
				"joint",	# str (Option from: ['joint', 'separate', 'vae'])
								in 'Refiner swap method' Dropdown component
				0,	# int | float (numeric value between 0.0 and 1.0)
								in 'Softness of ControlNet' Slider component
				True,	# bool in 'Enabled' Checkbox component
				0,	# int | float (numeric value between 0 and 2)
								in 'B1' Slider component
				0,	# int | float (numeric value between 0 and 2)
								in 'B2' Slider component
				0,	# int | float (numeric value between 0 and 4)
								in 'S1' Slider component
				0,	# int | float (numeric value between 0 and 4)
								in 'S2' Slider component
				True,	# bool in 'Debug Inpaint Preprocessing' Checkbox component
				True,	# bool in 'Disable initial latent in inpaint' Checkbox component
				"None",	# str (Option from: ['None', 'v1', 'v2.5', 'v2.6'])
								in 'Inpaint Engine' Dropdown component
				0,	# int | float (numeric value between 0.0 and 1.0)
								in 'Inpaint Denoising Strength' Slider component
				0,	# int | float (numeric value between 0.0 and 1.0)
								in 'Inpaint Respective Field' Slider component
				True,	# bool in 'Enable Mask Upload' Checkbox component
				True,	# bool in 'Invert Mask' Checkbox component
				-64,	# int | float (numeric value between -64 and 64)
								in 'Mask Erode or Dilate' Slider component
				True,	# bool in 'Save Metadata to Images' Checkbox component
				"fooocus",	# str in 'Metadata Scheme' Radio component
				"https://raw.githubusercontent.com/gradio-app/gradio/main/test/test_files/bus.png",	# str (filepath or URL to image)
								in 'Image' Image component
				0,	# int | float (numeric value between 0.0 and 1.0)
								in 'Stop At' Slider component
				0,	# int | float (numeric value between 0.0 and 2.0)
								in 'Weight' Slider component
				"ImagePrompt",	# str in 'Type' Radio component
				"https://raw.githubusercontent.com/gradio-app/gradio/main/test/test_files/bus.png",	# str (filepath or URL to image)
								in 'Image' Image component
				0,	# int | float (numeric value between 0.0 and 1.0)
								in 'Stop At' Slider component
				0,	# int | float (numeric value between 0.0 and 2.0)
								in 'Weight' Slider component
				"ImagePrompt",	# str in 'Type' Radio component
				"https://raw.githubusercontent.com/gradio-app/gradio/main/test/test_files/bus.png",	# str (filepath or URL to image)
								in 'Image' Image component
				0,	# int | float (numeric value between 0.0 and 1.0)
								in 'Stop At' Slider component
				0,	# int | float (numeric value between 0.0 and 2.0)
								in 'Weight' Slider component
				"ImagePrompt",	# str in 'Type' Radio component
				"https://raw.githubusercontent.com/gradio-app/gradio/main/test/test_files/bus.png",	# str (filepath or URL to image)
								in 'Image' Image component
				0,	# int | float (numeric value between 0.0 and 1.0)
								in 'Stop At' Slider component
				0,	# int | float (numeric value between 0.0 and 2.0)
								in 'Weight' Slider component
				"hello",	# str in 'Type' Radio component
				fn_index=41
)
print(result)
Error
  File "<ipython-input-28-095d75774bec>", line 10
    "704×1408 <span style="color: grey;"> ∣ 1:2</span>",	# str in 'Aspect Ratios' Radio component
    ^
SyntaxError: invalid syntax. Perhaps you forgot a comma?

**Reply by t1hq** [reply]
python requires more care on code format, in your code for example:
```python
0, # int | float (numeric value between 0.0 and 1.0)
# this line is also comment --------- in 'Stop At' Slider component
```
simply copy the code from sample does not work.
i don't think it is a good idea to fill the page with a long long code,

but it seems that you don't care, here's a more clear demo:
```python
# copied from https://tricks.one/post/get-fooocus-gradio-api-working-for-uov/
from gradio_client import Client

client = Client("http://127.0.0.1:7865/")

# Textbox: string
# Checkbox: bool
# Radio: string
# Dropdown: string
# Slider: float | int
# Checkboxgroup: list[string]
# Image: filepath or URL to image

result = client.predict(
	False,	# 'Generate Image Grid for Each Batch' Checkbox
	"",	# 'Prompt' Textbox
	"",	# 'Negative Prompt' Textbox
        # 'Selected Styles' Checkboxgroup
	["Fooocus V2", "Fooocus Semi Realistic", "Fooocus Masterpiece"],
	"Speed",	# 'Performance' Radio
	"1280×768",	# 'Aspect Ratios' Radio
	1,		# 'Image Number' Slider [1,32]
	"png",	# 'Output Format' Radio
	"0",	# 'Seed' Textbox
        False,	# 'Read wildcards in order' Checkbox
	2,		# 'Image Sharpness' Slider [0.0,30.0]
	7,		#  'Guidance Scale' Slider [1.0,30.0]
	"animaPencilXL_v100.safetensors",	# 'Base Model (SDXL only)' Dropdown
	"None",	# 'Refiner (SDXL or SD 1.5)' Dropdown
	0.5,	# 'Refiner Switch At' Slider [0.1,1.0]
	True,	# 'LoRA 1 Enable' Checkbox
	"None",	# 'LoRA 1' Dropdown
	1,	# 'LoRA 1 Weight' Slider [-2,2]
	True,	# 'LoRA 2 Enable' Checkbox
	"None",	# 'LoRA 2' Dropdown
	1,		# 'LoRA 2 Weight' Slider [-2,2]
	True,	# 'LoRA 3 Enable' Checkbox
	"None",	# 'LoRA 3' Dropdown
	1,		# 'LoRA 3 Weight' Slider [-2,2]
	True,	# 'LoRA 4 Enable' Checkbox
	"None",	# 'LoRA 4' Dropdown
	1,		# 'LoRA 4 Weight' Slider [-2,2]
	True,	# 'LoRA 5 Enable' Checkbox
	"None",	# 'LoRA 5' Dropdown component
	1,		# 'LoRA 5 Weight' Slider [-2,2]
	True,	#'Input Image' Checkbox
	"uov",		# 'current tab' Textbox ["uov", "ip", "inpaint"]
	"Vary (Subtle)",	# 'Upscale or Variation:' Radio
	"20240322214141.jpg",			# 'Upscale or Variation' Image
	["Left"],	# 'Outpaint Direction' Checkboxgroup
	"",		# 'Inpaint or Outpaint' Image
	"",		# 'Inpaint Additional Prompt' Textbox
	"",		# 'Inpaint or Outpaint Mask Upload' Image
	True,	# 'Developer Debug Mode: Disable Preview' Checkbox
	True,	# 'Developer Debug Mode: Disable Intermediate Results' Checkbox
	True,	# 'Developer Debug Mode: Disable seed increment' Checkbox
	1.5,	# 'Developer Debug Mode: Positive ADM Guidance Scaler' Slider [0.1,3.0]
	0.8,	# 'Developer Debug Mode: Negative ADM Guidance Scaler' Slider [0.1,3.0]
	0.3,	# 'Developer Debug Mode: ADM Guidance End At Step' Slider [0.0,1.0]
	7,		# 'Developer Debug Mode: CFG Mimicking from TSNR' Slider [1.0,30.0]
	"dpmpp_2m_sde_gpu",	# 'Developer Debug Mode: Sampler' Dropdown
	"karras",	# 'Developer Debug Mode: Scheduler' Dropdown
	-1,	# 'Developer Debug Mode: Forced Overwrite of Sampling Step' Slider [-1,200]
	-1,	# 'Developer Debug Mode: Forced Overwrite of Refiner Switch Step' Slider [-1,200]
	-1,	# 'Developer Debug Mode: Forced Overwrite of Generating Width' Slider [-1,2048]
	-1,	# 'Developer Debug Mode: Forced Overwrite of Generating Height' Slider [-1,2048]
	-1,	# 'Developer Debug Mode: Forced Overwrite of Denoising Strength of "Vary"' Slider [-1,1.0]
	-1,	# 'Developer Debug Mode: Forced Overwrite of Denoising Strength of "Upscale"' Slider [-1,1.0]
	False,	# 'Developer Debug Mode: Mixing Image Prompt and Vary/Upscale' Checkbox
	False,	# 'Developer Debug Mode: Mixing Image Prompt and Inpaint' Checkbox
	False,	# 'Developer Debug Mode: Debug Preprocessors' Checkbox
	False,	# 'Developer Debug Mode: Skip Preprocessors' Checkbox
	64,		# 'Developer Debug Mode: Canny Low Threshold' Slider [1,255]
	128,	# 'Developer Debug Mode: Canny High Threshold' Slider [1,255]
	"joint",	# 'Developer Debug Mode: Refiner swap method' Dropdown
	0.25,	# 'Developer Debug Mode: Softness of ControlNet' Slider [0.0,1.0]
	False,	# 'Developer Debug Mode: FreeU Enabled' Checkbox
	1.01,	# 'Developer Debug Mode: FreeU B1' Slider [0,2]
	1.02,	# 'Developer Debug Mode: FreeU B2' Slider [0,2]
	0.99,	# 'Developer Debug Mode: FreeU S1' Slider [0,4]
	0.95,	# 'Developer Debug Mode: FreeU S2' Slider [0,4]
	False,	# 'Developer Debug Mode: Debug Inpaint Preprocessing' Checkbox
	False,	# 'Developer Debug Mode: Disable initial latent in inpaint' Checkbox
	"v2.6",	# 'Developer Debug Mode: Inpaint Engine' Dropdown
	1,		# 'Developer Debug Mode: Inpaint Denoising Strength' Slider [0.0,1.0]
	0.618,	# 'Developer Debug Mode: Inpaint Respective Field' Slider [0.0,1.0]
	False,	# 'Developer Debug Mode: Enable Mask Upload' Checkbox
	False,	# 'Developer Debug Mode: Invert Mask' Checkbox
	0,		# 'Developer Debug Mode: Mask Erode or Dilate' Slider [-64,64]
	False,	# 'Developer Debug Mode: Save Metadata to Images' Checkbox
	"fooocus",	# 'Metadata Scheme' Radio
	"",				# 'Image Prompt Image 1' Image component
	0,				# 'Image Prompt Image 1 Stop At' Slider [0.0-1.0]
	0,				# 'Image Prompt Image 1 Weight' Slider [0.0-2.0]
	"ImagePrompt",	# 'Image Prompt Image 1 Type' Radio
	"",				# 'Image Prompt Image 2' Image component
	0,				# 'Image Prompt Image 2 Stop At' Slider [0.0-1.0]
	0,				# 'Image Prompt Image 2 Weight' Slider [0.0-2.0]
	"ImagePrompt",	# 'Image Prompt Image 2 Type' Radio
	"",				# 'Image Prompt Image 3' Image component
	0,				# 'Image Prompt Image 3 Stop At' Slider [0.0-1.0]
	0,				# 'Image Prompt Image 3 Weight' Slider [0.0-2.0]
	"ImagePrompt",	# 'Image Prompt Image 3 Type' Radio
	"",				# 'Image Prompt Image 4' Image component
	0,				# 'Image Prompt Image 4 Stop At' Slider [0.0-1.0]
	0,				# 'Image Prompt Image 4 Weight' Slider [0.0-2.0]
	"ImagePrompt",	# 'Image Prompt Image 4 Type' Radio
	fn_index=40
)
print(result)
result= client.predict(fn_index=41)
```

**Reply by ishangrotra** [reply]
hey sorry about messy code  i am sorta new so i am not sure about the best practices so please forgive me 
thank you for providing me the code just one more thing if i use the gradio live link it doesnt work you have any idea how i can use the gradio live link instead of http://127.0.0.1:7865/ ? 
thanks again for taking time and replying

**Reply by dentroai** [reply]
This seemed to work in March, however I tried it yesterday and got the error `Expected 1 arguments, got 91`. 

Anybody knows how to do it nowdays?

**Reply by dinasenpai** [reply]
Hii, I am facing the same ValueError: ` Expected 1 arguments, got 92`, Please revert back if found any solution.

**Reply by dentroai** [reply]
Hi I switched over to use https://github.com/mrhan1993/Fooocus-API when I need to use the api.

The instructions in the readme aren't that clear, but here's how to do it:
1. Clone the fooocus-api repo (`git clone https://github.com/mrhan1993/Fooocus-API.git`)
2. inside the cloned folder create a new virtual environment and activate it (`python3 -m venv .venv; source .venv/bin/activate`)
3. install the requirements.txt (`pip install -r requirements.txt`) 
4. if you already have Fooocus installed on your machine, go to the Fooocus folder and look for the `config.txt` file. Copy this file and paste it into the root folder of Fooocus-API
5. run Fooocus-API (inside the Fooocus-API root folder run `python3 main.py`)
6. go to http://localhost:8888/docs to see the swagger specification of the API
7. Generate an image, e.g.:

```
url = http://localhost:8888/v2/generation/text-to-image-with-ip # or e.g. "http://localhost:8888/v2/generation/image-prompt"

# Request headers
headers = {"accept": "image/png", "Content-Type": "application/json"}

# Request payload
payload = {
    # "prompt": "a happy customer of DentroAI who can't believe how much more productive his middle sized business is thanks to Dentro",
    "negative_prompt": "",
    "style_selections": [
        "Fooocus V2",
        # "Fooocus Enhance",
        # "Fooocus Sharp"
    ],
    "performance_selection": "Speed",
    "aspect_ratios_selection": "896*1152",
    "image_number": 1,
    "image_seed": -1,
    "sharpness": 2,
    "guidance_scale": 4,
    "base_model_name": "juggernautXL_v8Rundiffusion.safetensors",
    "base_model_name": "juggernautXL_v8Rundiffusion.safetensors",
    "refiner_switch": 0.5,
    "loras": [
        {"enabled": True, "model_name": "None", "weight": 1},
        {"enabled": True, "model_name": "None", "weight": 1},
        {"enabled": True, "model_name": "None", "weight": 1},
        {"enabled": True, "model_name": "None", "weight": 1},
        {"enabled": True, "model_name": "None", "weight": 1},
    ],
    "advanced_params": {
        "adaptive_cfg": 7,
        "adm_scaler_end": 0.3,
        "adm_scaler_negative": 0.8,
        "adm_scaler_positive": 1.5,
        "black_out_nsfw": False,
        "canny_high_threshold": 128,
        "canny_low_threshold": 64,
        "clip_skip": 2,
        "controlnet_softness": 0.25,
        "debugging_cn_preprocessor": False,
        "debugging_dino": False,
        "debugging_enhance_masks_checkbox": False,
        "debugging_inpaint_preprocessor": False,
        "dino_erode_or_dilate": 0,
        "disable_intermediate_results": False,
        "disable_preview": False,
        "disable_seed_increment": False,
        "freeu_b1": 1.01,
        "freeu_b2": 1.02,
        "freeu_enabled": False,
        "freeu_s1": 0.99,
        "freeu_s2": 0.95,
        "inpaint_advanced_masking_checkbox": True,
        "inpaint_disable_initial_latent": False,
        "inpaint_engine": "v2.6",
        "inpaint_erode_or_dilate": 0,
        "inpaint_respective_field": 1,
        "inpaint_strength": 1,
        "invert_mask_checkbox": False,
        "mixing_image_prompt_and_inpaint": False,
        "mixing_image_prompt_and_vary_upscale": False,
        "overwrite_height": -1,
        "overwrite_step": -1,
        "overwrite_switch": -1,
        "overwrite_upscale_strength": -1,
        "overwrite_vary_strength": -1,
        "overwrite_width": -1,
        "refiner_swap_method": "joint",
        "sampler_name": "dpmpp_2m_sde_gpu",
        "scheduler_name": "karras",
        "skipping_cn_preprocessor": False,
        "vae_name": "Default (model)",
    },
    "save_meta": True,
    "meta_scheme": "fooocus",
    "save_extension": "png",
    "save_name": "",
    "read_wildcards_in_order": False,
    "require_base64": False,
    "async_process": False,
    "webhook_url": "",
    "image_prompts": [],
}

response = requests.post(url, json=payload, headers=headers)
# Save the generated image
with open("happy_customer_of_dentroAI.png", "wb") as f:
    f.write(response.content)
```

Things to know:
- you don't need to run your other Fooocus instance at the same time while running Fooocus-API
- for manual image generation I still use the normal Fooocus, just running it as normal

**Reply by t1hq** [reply]
the api changed from 2.3.0 to 2.5.5, it looks like the key func is now `fn 67` and `fn 68`, but more params are added to `fn 67`, the params need reviewing if you want to use fooocus 2.5.5 api

**Reply by dinasenpai** [reply]
Thanks for the detailed approach, worked like a charm for me.

**Comment by mashb1t** [comment]
also see https://github.com/lllyasviel/Fooocus/discussions/2608

**Reply by ishangrotra** [reply]
hey thanks for taking time i wanted to ask will the provided code work in the link for gradio live link ?
or just for the local link 
example gradio live link :  https://4dc12345678.gradio.live/

**Reply by saibabu10** [reply]
I have a question how to connect with the javascript api ? please send me the code if you know

**Reply by mashb1t** [reply]
docs see https://www.gradio.app/guides/getting-started-with-the-js-client, same parameters and endpoint as above

**Reply by ijash** [reply]
> I have a question how to connect with the javascript api ? please send me the code if you know

```
import { Client } from "@gradio/client";

const app = await Client.connect("http://127.0.0.1:7865/");
const result = await app.predict(41, [true]);
console.log(result.data);
```
![image](https://github.com/user-attachments/assets/cea0aafc-2329-45d3-b442-0c1b4123c164)

i have successfully call the api up to that point. the rest remains unknown.

i just want generate simple image with default settings as the starting point. can anyone help with that?

**Reply by jiggly6969** [reply]
> > I have a question how to connect with the javascript api ? please send me the code if you know
> 
> ```
> import { Client } from "@gradio/client";
> 
> const app = await Client.connect("http://127.0.0.1:7865/");
> const result = await app.predict(41, [true]);
> console.log(result.data);
> ```
> 
> ![image](https://private-user-images.githubusercontent.com/30673286/421356106-cea0aafc-2329-45d3-b442-0c1b4123c164.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTY0OTYwMjYsIm5iZiI6MTc1NjQ5NTcyNiwicGF0aCI6Ii8zMDY3MzI4Ni80MjEzNTYxMDYtY2VhMGFhZmMtMjMyOS00NWQzLWI0NDItMGMxYjQxMjNjMTY0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODI5VDE5Mjg0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdhMjllZGQzNmU0NWJhYWIxMWUwMmM2MjJjMTQ2N2Q1Y2ZhYmRkYjg1Zjk4MzU5YWE1OGIwNjkxYmQ4NDY0OTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.aEQvFpKvQZcADCExrzA5rd3ICNlWazA4ipW20HCNTPE)
> 
> i have successfully call the api up to that point. the rest remains unknown.
> 
> i just want generate simple image with default settings as the starting point. can anyone help with that?

Did you get it?, I am struggling alot, really need the perfect default package to just edit the prompt, I am so tired
